File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int n = numbers.size() ; 
        int sum =0 ; 
        int i =0 , j = n-1 ; 
        while (i<=j){
            sum = numbers[i] + numbers[j] ; 
            if (sum > target ){
                j-- ; 
            }
            else if (sum < target){
                i++ ; 
            }
           else break ; 
        }
        return {i+1 , j+1} ; 
    }
};

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
from typing import List

class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        i, j = 0, len(numbers) - 1

        while i <= j:
            sum_ = numbers[i] + numbers[j]
            if sum_ > target:
                j -= 1
            elif sum_ < target:
                i += 1
            else:
                break
        
        return [i + 1, j + 1]
